{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "08fbbb66-45e0-4504-a7b5-57b446b1121a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from alphagenome.data import genome\n",
    "from alphagenome.models import dna_client, variant_scorers\n",
    "import pandas as pd\n",
    "from tqdm import tqdm\n",
    "from dotenv import load_dotenv\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "load_dotenv()\n",
    "ALPHAGENOME_KEY=os.getenv('ALPHAGENOME_KEY')\n",
    "save_predictions = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "263f0023-5b86-43b8-867f-91154ad6b1f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>A1FREQ_CASES</th>\n",
       "      <th>A1FREQ_CONTROLS</th>\n",
       "      <th>N</th>\n",
       "      <th>N_CASES</th>\n",
       "      <th>N_CONTROLS</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26205065</td>\n",
       "      <td>6:26205065:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.468846</td>\n",
       "      <td>0.490179</td>\n",
       "      <td>0.496948</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.066559</td>\n",
       "      <td>0.012493</td>\n",
       "      <td>28.3848</td>\n",
       "      <td>7.00242</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26210918</td>\n",
       "      <td>6:26210918:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.273908</td>\n",
       "      <td>0.289524</td>\n",
       "      <td>0.290326</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.075221</td>\n",
       "      <td>0.013859</td>\n",
       "      <td>29.4583</td>\n",
       "      <td>7.24311</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26214245</td>\n",
       "      <td>6:26214245:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.474028</td>\n",
       "      <td>0.495378</td>\n",
       "      <td>0.502440</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.066576</td>\n",
       "      <td>0.012478</td>\n",
       "      <td>28.4681</td>\n",
       "      <td>7.02110</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26225804</td>\n",
       "      <td>6:26225804:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.284913</td>\n",
       "      <td>0.301078</td>\n",
       "      <td>0.301990</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.073885</td>\n",
       "      <td>0.013698</td>\n",
       "      <td>29.0931</td>\n",
       "      <td>7.16125</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26233159</td>\n",
       "      <td>6:26233159:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.274226</td>\n",
       "      <td>0.291707</td>\n",
       "      <td>0.290663</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.079009</td>\n",
       "      <td>0.013844</td>\n",
       "      <td>32.5706</td>\n",
       "      <td>7.93952</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49274119</td>\n",
       "      <td>20:49274119:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.391124</td>\n",
       "      <td>0.373098</td>\n",
       "      <td>0.414567</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.077317</td>\n",
       "      <td>0.013166</td>\n",
       "      <td>34.4866</td>\n",
       "      <td>8.36734</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49275101</td>\n",
       "      <td>20:49275101:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.390888</td>\n",
       "      <td>0.372745</td>\n",
       "      <td>0.414317</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.077830</td>\n",
       "      <td>0.013168</td>\n",
       "      <td>34.9351</td>\n",
       "      <td>8.46739</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49278429</td>\n",
       "      <td>20:49278429:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.389233</td>\n",
       "      <td>0.371333</td>\n",
       "      <td>0.412563</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.077548</td>\n",
       "      <td>0.013177</td>\n",
       "      <td>34.6343</td>\n",
       "      <td>8.40029</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49291253</td>\n",
       "      <td>20:49291253:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.390944</td>\n",
       "      <td>0.372713</td>\n",
       "      <td>0.414377</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.077937</td>\n",
       "      <td>0.013169</td>\n",
       "      <td>35.0244</td>\n",
       "      <td>8.48732</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49297643</td>\n",
       "      <td>20:49297643:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.391015</td>\n",
       "      <td>0.372906</td>\n",
       "      <td>0.414452</td>\n",
       "      <td>275488</td>\n",
       "      <td>15579</td>\n",
       "      <td>259909</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.077592</td>\n",
       "      <td>0.013168</td>\n",
       "      <td>34.7212</td>\n",
       "      <td>8.41968</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>342 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM    GENPOS               ID ALLELE0 ALLELE1    A1FREQ  A1FREQ_CASES  \\\n",
       "0     chr6  26205065   6:26205065:G:A       G       A  0.468846      0.490179   \n",
       "1     chr6  26210918   6:26210918:G:T       G       T  0.273908      0.289524   \n",
       "2     chr6  26214245   6:26214245:T:C       T       C  0.474028      0.495378   \n",
       "3     chr6  26225804   6:26225804:C:G       C       G  0.284913      0.301078   \n",
       "4     chr6  26233159   6:26233159:A:G       A       G  0.274226      0.291707   \n",
       "..     ...       ...              ...     ...     ...       ...           ...   \n",
       "337  chr20  49274119  20:49274119:A:G       A       G  0.391124      0.373098   \n",
       "338  chr20  49275101  20:49275101:C:T       C       T  0.390888      0.372745   \n",
       "339  chr20  49278429  20:49278429:G:A       G       A  0.389233      0.371333   \n",
       "340  chr20  49291253  20:49291253:A:G       A       G  0.390944      0.372713   \n",
       "341  chr20  49297643  20:49297643:G:C       G       C  0.391015      0.372906   \n",
       "\n",
       "     A1FREQ_CONTROLS       N  N_CASES  N_CONTROLS TEST      BETA        SE  \\\n",
       "0           0.496948  275488    15579      259909  ADD  0.066559  0.012493   \n",
       "1           0.290326  275488    15579      259909  ADD  0.075221  0.013859   \n",
       "2           0.502440  275488    15579      259909  ADD  0.066576  0.012478   \n",
       "3           0.301990  275488    15579      259909  ADD  0.073885  0.013698   \n",
       "4           0.290663  275488    15579      259909  ADD  0.079009  0.013844   \n",
       "..               ...     ...      ...         ...  ...       ...       ...   \n",
       "337         0.414567  275488    15579      259909  ADD -0.077317  0.013166   \n",
       "338         0.414317  275488    15579      259909  ADD -0.077830  0.013168   \n",
       "339         0.412563  275488    15579      259909  ADD -0.077548  0.013177   \n",
       "340         0.414377  275488    15579      259909  ADD -0.077937  0.013169   \n",
       "341         0.414452  275488    15579      259909  ADD -0.077592  0.013168   \n",
       "\n",
       "       CHISQ   LOG10P  EXTRA  \n",
       "0    28.3848  7.00242    NaN  \n",
       "1    29.4583  7.24311    NaN  \n",
       "2    28.4681  7.02110    NaN  \n",
       "3    29.0931  7.16125    NaN  \n",
       "4    32.5706  7.93952    NaN  \n",
       "..       ...      ...    ...  \n",
       "337  34.4866  8.36734    NaN  \n",
       "338  34.9351  8.46739    NaN  \n",
       "339  34.6343  8.40029    NaN  \n",
       "340  35.0244  8.48732    NaN  \n",
       "341  34.7212  8.41968    NaN  \n",
       "\n",
       "[342 rows x 17 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load tsv file which only has variants above -log10p of 7.\n",
    "df = pd.read_csv('gwas1_log10p_gt_7.tsv', sep='\\t')\n",
    "df.CHROM = 'chr' + df.CHROM.astype(str)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "886e58bb-da30-4b3d-82bb-2b7036f09f96",
   "metadata": {},
   "outputs": [],
   "source": [
    "dna_model = dna_client.create(ALPHAGENOME_KEY)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c75fc248-270e-4e13-be82-f23fbcbf57be",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequence length and scorer selection\n",
    "\n",
    "sequence_length = '1MB'  # \"16KB\", \"100KB\", \"500KB\", \"1MB\"\n",
    "sequence_length = dna_client.SUPPORTED_SEQUENCE_LENGTHS[\n",
    "    f'SEQUENCE_LENGTH_{sequence_length}'\n",
    "]\n",
    "\n",
    "scorer_selections = {\n",
    "    'rna_seq': True,\n",
    "    'cage': False,\n",
    "    'procap': False,\n",
    "    'atac': False,\n",
    "    'dnase': False,\n",
    "    'chip_histone': False,\n",
    "    'chip_tf': False,\n",
    "    'polyadenylation': False,\n",
    "    'splice_sites': False,\n",
    "    'splice_site_usage': False,\n",
    "    'splice_junctions': False,\n",
    "}\n",
    "\n",
    "all_scorers = variant_scorers.RECOMMENDED_VARIANT_SCORERS\n",
    "selected_scorers = [\n",
    "    all_scorers[key]\n",
    "    for key in all_scorers\n",
    "    if scorer_selections.get(key.lower(), False)\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "817aee18-6fed-4ecf-8ff3-92ebdbc8f23c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 342/342 [06:31<00:00,  1.14s/it]\n"
     ]
    }
   ],
   "source": [
    "results = []\n",
    "for i, row in tqdm(df.iterrows(), total=len(df)):\n",
    "  variant = genome.Variant(\n",
    "      chromosome=str(row.CHROM),\n",
    "      position=int(row.GENPOS),\n",
    "      reference_bases=row.ALLELE0,\n",
    "      alternate_bases=row.ALLELE1,\n",
    "      name=row.ID,\n",
    "  )\n",
    "  interval = variant.reference_interval.resize(sequence_length)\n",
    "\n",
    "  variant_scores = dna_model.score_variant(\n",
    "      interval=interval,\n",
    "      variant=variant,\n",
    "      variant_scorers=selected_scorers,\n",
    "      organism=dna_client.Organism.HOMO_SAPIENS,\n",
    "  )\n",
    "  results.append(variant_scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e27cc84e-aaf9-40f8-9d37-93bb5a4ac5db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>scored_interval</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>junction_Start</th>\n",
       "      <th>junction_End</th>\n",
       "      <th>output_type</th>\n",
       "      <th>variant_scorer</th>\n",
       "      <th>...</th>\n",
       "      <th>biosample_name</th>\n",
       "      <th>biosample_type</th>\n",
       "      <th>biosample_life_stage</th>\n",
       "      <th>gtex_tissue</th>\n",
       "      <th>data_source</th>\n",
       "      <th>endedness</th>\n",
       "      <th>genetically_modified</th>\n",
       "      <th>raw_score</th>\n",
       "      <th>quantile_score</th>\n",
       "      <th>abs_quantile_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2095528</th>\n",
       "      <td>chr20:48905549:CT&gt;C</td>\n",
       "      <td>chr20:48381261-49429837:.</td>\n",
       "      <td>ENSG00000230758</td>\n",
       "      <td>SNAP23P1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>endothelial cell</td>\n",
       "      <td>in_vitro_differentiated_cells</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>single</td>\n",
       "      <td>False</td>\n",
       "      <td>1.833034e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609855</th>\n",
       "      <td>chr20:49012496:T&gt;TTTTG</td>\n",
       "      <td>chr20:48488208-49536784:.</td>\n",
       "      <td>ENSG00000226284</td>\n",
       "      <td>ARPC3P1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>MG63</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>child</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>1.973138e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3102464</th>\n",
       "      <td>chr20:49105400:TTTG&gt;T</td>\n",
       "      <td>chr20:48581113-49629689:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>endothelial cell of coronary artery</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-3.197962e-01</td>\n",
       "      <td>-9.999800e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3102480</th>\n",
       "      <td>chr20:49105400:TTTG&gt;T</td>\n",
       "      <td>chr20:48581113-49629689:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>dermis microvascular lymphatic vessel endothel...</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-4.071872e-01</td>\n",
       "      <td>-9.999800e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3102472</th>\n",
       "      <td>chr20:49105400:TTTG&gt;T</td>\n",
       "      <td>chr20:48581113-49629689:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>bladder microvascular endothelial cell</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-4.263840e-01</td>\n",
       "      <td>-9.999800e-01</td>\n",
       "      <td>9.999800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87802</th>\n",
       "      <td>chr6:26214245:T&gt;C</td>\n",
       "      <td>chr6:25689957-26738533:.</td>\n",
       "      <td>ENSG00000287050</td>\n",
       "      <td>ENSG00000287050</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>body of pancreas</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973938</th>\n",
       "      <td>chr6:26306054:T&gt;C</td>\n",
       "      <td>chr6:25781766-26830342:.</td>\n",
       "      <td>ENSG00000196866</td>\n",
       "      <td>H2AC7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>bronchial smooth muscle cell</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-2.384186e-07</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208546</th>\n",
       "      <td>chr6:26325805:C&gt;T</td>\n",
       "      <td>chr6:25801517-26850093:.</td>\n",
       "      <td>ENSG00000287050</td>\n",
       "      <td>ENSG00000287050</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>Ammon's horn</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>gtex</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-4.768372e-07</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2117894</th>\n",
       "      <td>chr20:48906783:C&gt;T</td>\n",
       "      <td>chr20:48382495-49431071:.</td>\n",
       "      <td>ENSG00000236874</td>\n",
       "      <td>ENSG00000236874</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>urinary bladder</td>\n",
       "      <td>tissue</td>\n",
       "      <td>embryonic</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-4.768372e-07</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3382849</th>\n",
       "      <td>chr20:49144319:A&gt;G</td>\n",
       "      <td>chr20:48620031-49668607:.</td>\n",
       "      <td>ENSG00000283529</td>\n",
       "      <td>ENSG00000283529</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>CD8-positive, alpha-beta T cell</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-2.861023e-06</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3800412 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     variant_id            scored_interval          gene_id  \\\n",
       "2095528     chr20:48905549:CT>C  chr20:48381261-49429837:.  ENSG00000230758   \n",
       "2609855  chr20:49012496:T>TTTTG  chr20:48488208-49536784:.  ENSG00000226284   \n",
       "3102464   chr20:49105400:TTTG>T  chr20:48581113-49629689:.  ENSG00000124212   \n",
       "3102480   chr20:49105400:TTTG>T  chr20:48581113-49629689:.  ENSG00000124212   \n",
       "3102472   chr20:49105400:TTTG>T  chr20:48581113-49629689:.  ENSG00000124212   \n",
       "...                         ...                        ...              ...   \n",
       "87802         chr6:26214245:T>C   chr6:25689957-26738533:.  ENSG00000287050   \n",
       "973938        chr6:26306054:T>C   chr6:25781766-26830342:.  ENSG00000196866   \n",
       "1208546       chr6:26325805:C>T   chr6:25801517-26850093:.  ENSG00000287050   \n",
       "2117894      chr20:48906783:C>T  chr20:48382495-49431071:.  ENSG00000236874   \n",
       "3382849      chr20:49144319:A>G  chr20:48620031-49668607:.  ENSG00000283529   \n",
       "\n",
       "               gene_name                           gene_type gene_strand  \\\n",
       "2095528         SNAP23P1                processed_pseudogene           -   \n",
       "2609855          ARPC3P1                processed_pseudogene           +   \n",
       "3102464            PTGIS                      protein_coding           -   \n",
       "3102480            PTGIS                      protein_coding           -   \n",
       "3102472            PTGIS                      protein_coding           -   \n",
       "...                  ...                                 ...         ...   \n",
       "87802    ENSG00000287050                              lncRNA           -   \n",
       "973938             H2AC7                      protein_coding           -   \n",
       "1208546  ENSG00000287050                              lncRNA           -   \n",
       "2117894  ENSG00000236874                              lncRNA           -   \n",
       "3382849  ENSG00000283529  transcribed_unprocessed_pseudogene           +   \n",
       "\n",
       "        junction_Start junction_End output_type  \\\n",
       "2095528           None         None     RNA_SEQ   \n",
       "2609855           None         None     RNA_SEQ   \n",
       "3102464           None         None     RNA_SEQ   \n",
       "3102480           None         None     RNA_SEQ   \n",
       "3102472           None         None     RNA_SEQ   \n",
       "...                ...          ...         ...   \n",
       "87802             None         None     RNA_SEQ   \n",
       "973938            None         None     RNA_SEQ   \n",
       "1208546           None         None     RNA_SEQ   \n",
       "2117894           None         None     RNA_SEQ   \n",
       "3382849           None         None     RNA_SEQ   \n",
       "\n",
       "                                      variant_scorer  ...  \\\n",
       "2095528  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "2609855  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3102464  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3102480  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3102472  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "...                                              ...  ...   \n",
       "87802    GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "973938   GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "1208546  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "2117894  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3382849  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "\n",
       "                                            biosample_name  \\\n",
       "2095528                                   endothelial cell   \n",
       "2609855                                               MG63   \n",
       "3102464                endothelial cell of coronary artery   \n",
       "3102480  dermis microvascular lymphatic vessel endothel...   \n",
       "3102472             bladder microvascular endothelial cell   \n",
       "...                                                    ...   \n",
       "87802                                     body of pancreas   \n",
       "973938                        bronchial smooth muscle cell   \n",
       "1208546                                       Ammon's horn   \n",
       "2117894                                    urinary bladder   \n",
       "3382849                    CD8-positive, alpha-beta T cell   \n",
       "\n",
       "                        biosample_type biosample_life_stage  \\\n",
       "2095528  in_vitro_differentiated_cells                adult   \n",
       "2609855                      cell_line                child   \n",
       "3102464                   primary_cell                adult   \n",
       "3102480                   primary_cell                adult   \n",
       "3102472                   primary_cell                adult   \n",
       "...                                ...                  ...   \n",
       "87802                           tissue                adult   \n",
       "973938                    primary_cell                adult   \n",
       "1208546                         tissue                adult   \n",
       "2117894                         tissue            embryonic   \n",
       "3382849                   primary_cell                adult   \n",
       "\n",
       "               gtex_tissue data_source endedness genetically_modified  \\\n",
       "2095528                         encode    single                False   \n",
       "2609855                         encode    paired                False   \n",
       "3102464                         encode    paired                False   \n",
       "3102480                         encode    paired                False   \n",
       "3102472                         encode    paired                False   \n",
       "...                    ...         ...       ...                  ...   \n",
       "87802                           encode    paired                False   \n",
       "973938                          encode    paired                False   \n",
       "1208546  Brain_Hippocampus        gtex    paired                False   \n",
       "2117894                         encode    paired                False   \n",
       "3382849                         encode    paired                False   \n",
       "\n",
       "            raw_score quantile_score abs_quantile_score  \n",
       "2095528  1.833034e-01   9.999800e-01       9.999800e-01  \n",
       "2609855  1.973138e-01   9.999800e-01       9.999800e-01  \n",
       "3102464 -3.197962e-01  -9.999800e-01       9.999800e-01  \n",
       "3102480 -4.071872e-01  -9.999800e-01       9.999800e-01  \n",
       "3102472 -4.263840e-01  -9.999800e-01       9.999800e-01  \n",
       "...               ...            ...                ...  \n",
       "87802    0.000000e+00   1.137757e-12       1.137757e-12  \n",
       "973938  -2.384186e-07   1.137757e-12       1.137757e-12  \n",
       "1208546 -4.768372e-07   1.137757e-12       1.137757e-12  \n",
       "2117894 -4.768372e-07   1.137757e-12       1.137757e-12  \n",
       "3382849 -2.861023e-06   1.137757e-12       1.137757e-12  \n",
       "\n",
       "[3800412 rows x 24 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_scores = variant_scorers.tidy_scores(results)\n",
    "df_scores['abs_quantile_score'] = abs(df_scores['quantile_score'])\n",
    "df_scores = df_scores.sort_values('abs_quantile_score', ascending=False)\n",
    "\n",
    "df_scores.to_csv('variant-scores_rnaseq_decodeme_significant_all-tissues.csv.gz', index=False, compression='gzip')\n",
    "\n",
    "df_scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ab7622a6-2c91-4f4e-accd-3b10c555518e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>scored_interval</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>junction_Start</th>\n",
       "      <th>junction_End</th>\n",
       "      <th>output_type</th>\n",
       "      <th>variant_scorer</th>\n",
       "      <th>...</th>\n",
       "      <th>biosample_name</th>\n",
       "      <th>biosample_type</th>\n",
       "      <th>biosample_life_stage</th>\n",
       "      <th>gtex_tissue</th>\n",
       "      <th>data_source</th>\n",
       "      <th>endedness</th>\n",
       "      <th>genetically_modified</th>\n",
       "      <th>raw_score</th>\n",
       "      <th>quantile_score</th>\n",
       "      <th>abs_quantile_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3146736</th>\n",
       "      <td>chr20:49109692:C&gt;CTTTT</td>\n",
       "      <td>chr20:48585404-49633980:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>1.050655e-01</td>\n",
       "      <td>9.999366e-01</td>\n",
       "      <td>9.999366e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284171</th>\n",
       "      <td>chr6:26328470:A&gt;AT</td>\n",
       "      <td>chr6:25804182-26852758:.</td>\n",
       "      <td>ENSG00000124557</td>\n",
       "      <td>BTN1A1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-9.972215e-02</td>\n",
       "      <td>-9.999220e-01</td>\n",
       "      <td>9.999220e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3283438</th>\n",
       "      <td>chr20:49132759:CCAAAA&gt;C</td>\n",
       "      <td>chr20:48608473-49657049:.</td>\n",
       "      <td>ENSG00000124228</td>\n",
       "      <td>DDX27</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>7.458559e-02</td>\n",
       "      <td>9.999164e-01</td>\n",
       "      <td>9.999164e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3103176</th>\n",
       "      <td>chr20:49105400:TTTG&gt;T</td>\n",
       "      <td>chr20:48581113-49629689:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-8.859038e-02</td>\n",
       "      <td>-9.998872e-01</td>\n",
       "      <td>9.998872e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3460368</th>\n",
       "      <td>chr20:49153402:T&gt;TA</td>\n",
       "      <td>chr20:48629114-49677690:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>7.188678e-02</td>\n",
       "      <td>9.998845e-01</td>\n",
       "      <td>9.998845e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3016056</th>\n",
       "      <td>chr20:49100868:T&gt;C</td>\n",
       "      <td>chr20:48576580-49625156:.</td>\n",
       "      <td>ENSG00000124212</td>\n",
       "      <td>PTGIS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-2.980232e-06</td>\n",
       "      <td>-3.459415e-02</td>\n",
       "      <td>3.459415e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2494273</th>\n",
       "      <td>chr20:48955014:C&gt;G</td>\n",
       "      <td>chr20:48430726-49479302:.</td>\n",
       "      <td>ENSG00000124201</td>\n",
       "      <td>ZNFX1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-2.801418e-06</td>\n",
       "      <td>-3.459415e-02</td>\n",
       "      <td>3.459415e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1040886</th>\n",
       "      <td>chr6:26317156:C&gt;T</td>\n",
       "      <td>chr6:25792868-26841444:.</td>\n",
       "      <td>ENSG00000124564</td>\n",
       "      <td>SLC17A3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-1.430511e-06</td>\n",
       "      <td>-2.306788e-02</td>\n",
       "      <td>2.306788e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882492</th>\n",
       "      <td>chr6:26302950:T&gt;G</td>\n",
       "      <td>chr6:25778662-26827238:.</td>\n",
       "      <td>ENSG00000181315</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>-8.344650e-07</td>\n",
       "      <td>-1.153548e-02</td>\n",
       "      <td>1.153548e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023407</th>\n",
       "      <td>chr17:52220547:A&gt;G</td>\n",
       "      <td>chr17:51696259-52744835:.</td>\n",
       "      <td>ENSG00000154975</td>\n",
       "      <td>CA10</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>...</td>\n",
       "      <td>brain</td>\n",
       "      <td>tissue</td>\n",
       "      <td>adult</td>\n",
       "      <td></td>\n",
       "      <td>encode</td>\n",
       "      <td>paired</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.137757e-12</td>\n",
       "      <td>1.137757e-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4398 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      variant_id            scored_interval          gene_id  \\\n",
       "3146736   chr20:49109692:C>CTTTT  chr20:48585404-49633980:.  ENSG00000124212   \n",
       "1284171       chr6:26328470:A>AT   chr6:25804182-26852758:.  ENSG00000124557   \n",
       "3283438  chr20:49132759:CCAAAA>C  chr20:48608473-49657049:.  ENSG00000124228   \n",
       "3103176    chr20:49105400:TTTG>T  chr20:48581113-49629689:.  ENSG00000124212   \n",
       "3460368      chr20:49153402:T>TA  chr20:48629114-49677690:.  ENSG00000124212   \n",
       "...                          ...                        ...              ...   \n",
       "3016056       chr20:49100868:T>C  chr20:48576580-49625156:.  ENSG00000124212   \n",
       "2494273       chr20:48955014:C>G  chr20:48430726-49479302:.  ENSG00000124201   \n",
       "1040886        chr6:26317156:C>T   chr6:25792868-26841444:.  ENSG00000124564   \n",
       "882492         chr6:26302950:T>G   chr6:25778662-26827238:.  ENSG00000181315   \n",
       "2023407       chr17:52220547:A>G  chr17:51696259-52744835:.  ENSG00000154975   \n",
       "\n",
       "        gene_name       gene_type gene_strand junction_Start junction_End  \\\n",
       "3146736     PTGIS  protein_coding           -           None         None   \n",
       "1284171    BTN1A1  protein_coding           +           None         None   \n",
       "3283438     DDX27  protein_coding           +           None         None   \n",
       "3103176     PTGIS  protein_coding           -           None         None   \n",
       "3460368     PTGIS  protein_coding           -           None         None   \n",
       "...           ...             ...         ...            ...          ...   \n",
       "3016056     PTGIS  protein_coding           -           None         None   \n",
       "2494273     ZNFX1  protein_coding           -           None         None   \n",
       "1040886   SLC17A3  protein_coding           -           None         None   \n",
       "882492     ZNF322  protein_coding           -           None         None   \n",
       "2023407      CA10  protein_coding           -           None         None   \n",
       "\n",
       "        output_type                               variant_scorer  ...  \\\n",
       "3146736     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "1284171     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3283438     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3103176     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "3460368     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "...             ...                                          ...  ...   \n",
       "3016056     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "2494273     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "1040886     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "882492      RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "2023407     RNA_SEQ  GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
       "\n",
       "        biosample_name biosample_type biosample_life_stage gtex_tissue  \\\n",
       "3146736          brain         tissue                adult               \n",
       "1284171          brain         tissue                adult               \n",
       "3283438          brain         tissue                adult               \n",
       "3103176          brain         tissue                adult               \n",
       "3460368          brain         tissue                adult               \n",
       "...                ...            ...                  ...         ...   \n",
       "3016056          brain         tissue                adult               \n",
       "2494273          brain         tissue                adult               \n",
       "1040886          brain         tissue                adult               \n",
       "882492           brain         tissue                adult               \n",
       "2023407          brain         tissue                adult               \n",
       "\n",
       "        data_source endedness genetically_modified     raw_score  \\\n",
       "3146736      encode    paired                False  1.050655e-01   \n",
       "1284171      encode    paired                False -9.972215e-02   \n",
       "3283438      encode    paired                False  7.458559e-02   \n",
       "3103176      encode    paired                False -8.859038e-02   \n",
       "3460368      encode    paired                False  7.188678e-02   \n",
       "...             ...       ...                  ...           ...   \n",
       "3016056      encode    paired                False -2.980232e-06   \n",
       "2494273      encode    paired                False -2.801418e-06   \n",
       "1040886      encode    paired                False -1.430511e-06   \n",
       "882492       encode    paired                False -8.344650e-07   \n",
       "2023407      encode    paired                False  0.000000e+00   \n",
       "\n",
       "        quantile_score abs_quantile_score  \n",
       "3146736   9.999366e-01       9.999366e-01  \n",
       "1284171  -9.999220e-01       9.999220e-01  \n",
       "3283438   9.999164e-01       9.999164e-01  \n",
       "3103176  -9.998872e-01       9.998872e-01  \n",
       "3460368   9.998845e-01       9.998845e-01  \n",
       "...                ...                ...  \n",
       "3016056  -3.459415e-02       3.459415e-02  \n",
       "2494273  -3.459415e-02       3.459415e-02  \n",
       "1040886  -2.306788e-02       2.306788e-02  \n",
       "882492   -1.153548e-02       1.153548e-02  \n",
       "2023407   1.137757e-12       1.137757e-12  \n",
       "\n",
       "[4398 rows x 24 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ontologies = ['UBERON:0000955']\n",
    "gene_types = ['protein_coding']\n",
    "df_scores_subset = df_scores[df_scores['ontology_curie'].isin(ontologies) & df_scores['gene_type'].isin(gene_types)]\n",
    "\n",
    "if save_predictions:\n",
    "  df_scores_subset.to_csv('variant-scores_rnaseq_decodeme_significant_brain_protein-coding.csv.gz', index=False, compression='gzip')\n",
    "\n",
    "df_scores_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a1319de-837b-4fe6-93ee-3c673e776812",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
